### PR TITLE
support requeueing

### PR DESCRIPTION
**Pull request by QuentinBisson**
Closes https://github.com/giantswarm/giantswarm/issues/29466 and closes https://github.com/giantswarm/giantswarm/issues/29879

### CODE REVIEW

File: pkg/logging-reconciler/logging_reconciler.go [hunk 1]
```diff
@@ -24,22 +24,13 @@ type LoggingReconciler struct {
...
 func (l *LoggingReconciler) Reconcile(ctx context.Context, lc loggedcluster.Interface) (result ctrl.Result, err error) {
-	loggingEnabled := common.IsLoggingEnabled(lc)
-
-	if loggingEnabled {
-		// TODO: handle returned ctrl.Result
-		_, err = l.reconcileCreate(ctx, lc)
-		if err != nil {
-			return ctrl.Result{}, errors.WithStack(err)
-		}
+	if common.IsLoggingEnabled(lc) {
```
Comment (QuentinBisson) [inline pkg/logging-reconciler/logging_reconciler.go:27]: Basic refactoring

File: pkg/logging-reconciler/logging_reconciler.go [hunk 2]
```diff
@@ -62,10 +53,9 @@ func (l *LoggingReconciler) reconcileCreate(ctx context.Context, lc loggedcluste
 
 	// Call all reconcilers ReconcileCreate methods.
 	for _, reconciler := range l.Reconcilers {
-		// TODO(theo): add handling for returned ctrl.Result value.
-		_, err := reconciler.ReconcileCreate(ctx, lc)
-		if err != nil {
-			return ctrl.Result{}, errors.WithStack(err)
+		result, err := reconciler.ReconcileCreate(ctx, lc)
+		if err != nil || !result.IsZero() {
```
Comment (QuentinBisson) [inline pkg/logging-reconciler/logging_reconciler.go:57]: If there is one error or the returned result is not empty (either requeue or requeueafter then we return to close the reconciliation loop)

**Comment by QuentinBisson** [comment]
Here are the requeing logs:

```
 k logs -n monitoring logging-operator-94c894b99-s52ql -f | grep requeuing
2024-02-20T16:59:23Z	INFO	grafana-agent-secret - app not found, requeuing	{"controller": "cluster", "controllerGroup": "cluster.x-k8s.io", "controllerKind": "Cluster", "Cluster": {"name":"sk43q","namespace":"org-quentin"}, "namespace": "org-quentin", "name": "sk43q", "reconcileID": "b7681f75-d763-4799-a86f-b83f1235d7dc"}
2024-02-20T16:59:24Z	INFO	grafana-agent-secret - app not found, requeuing	{"controller": "cluster", "controllerGroup": "cluster.x-k8s.io", "controllerKind": "Cluster", "Cluster": {"name":"sk43q","namespace":"org-quentin"}, "namespace": "org-quentin", "name": "sk43q", "reconcileID": "cf4ce982-aa0b-44ab-aab5-fdc18ddd8043"}
2024-02-20T16:59:55Z	ERROR	failed to remove finalizer from logger cluster, requeuing	{"controller": "cluster", "controllerGroup": "cluster.x-k8s.io", "controllerKind": "Cluster", "Cluster": {"name":"sk43q","namespace":"org-quentin"}, "namespace": "org-quentin", "name": "sk43q", "reconcileID": "694f1daf-829e-4543-b4c4-fc69a5b3c2b7", "finalizer": "giantswarm.io/logging-operator", "error": "Operation cannot be fulfilled on clusters.cluster.x-k8s.io \"sk43q\": the object has been modified; please apply your changes to the latest version and try again"}
```

Those are now not considered reconciliation errors as they will eventually fix themselves (usually fixed on the second run)

**Review by marieroque — APPROVED** [review]
Great :-)
Tested on `gaia`

**Pull request merged**